### PR TITLE
[15.0][FIX] stock_inventory: prevent error when applying inventory without quantities

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Just create an inventory adjustment for all products and select all the quants and click on "apply".

```
  File "/opt/odoo/auto/addons/stock_inventory/models/stock_quant.py", line 54, in _apply_inventory
    if adjustment and self.env.company.stock_inventory_auto_complete:
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 643, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
UnboundLocalError: local variable 'adjustment' referenced before assignment
```

cc @ForgeFlow

### CODE REVIEW

**Review by JordiMForgeFlow — APPROVED** [review]

File: stock_inventory/models/stock_quant.py [hunk 1]
```diff
@@ -9,6 +9,7 @@ class StockQuant(models.Model):
     def _apply_inventory(self):
         res = super()._apply_inventory()
         record_moves = self.env["stock.move.line"]
+        adjustment = False
```
Comment (rousseldenis) [inline stock_inventory/models/stock_quant.py]:
```suggestion
        adjustment = self.env["stock.inventory"].browse()
```

is semantically more correct (False is the result of the evaluation of a void recordset)

Comment (AaronHForgeFlow) [reply stock_inventory/models/stock_quant.py]: Sure! Just did that. Thanks

**Review by rousseldenis — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
@AaronHForgeFlow could you fwport to 16.0?

**Pull request merged**